testing/e2e/tests/types.go

```diff
@@ -3,25 +3,55 @@ package tests
 import (
 	"context"
 
-	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
+	"k8s.io/client-go/rest"
 )
 
-type RunStrategy int
-
-const (
-	// Parallel runs the test at the same time as other tests
-	Parallel RunStrategy = iota
-	// Sequential runs the test alone
-	Sequential
-)
+type test interface {
+	GetName() string
+	Run(ctx context.Context, config *rest.Config) error
+}
 
 // T is an interface for a single test
 type T interface {
-	GetName() string
-	GetRunStrategy() RunStrategy
-	Run(ctx context.Context) error
-	ShouldRun(infra infra.Provisioned) bool
+	// GetOperatorConfigs returns a slice of OperatorConfig structs that should be used for this test.
+	// All OperatorConfigs that are compatible should be returned.
+	GetOperatorConfigs() []manifests.OperatorConfig
+	test
 }
 
 // Ts is a slice of T
 type Ts []T
+
+type ordered []testsWithRunInfo
+
+type testsWithRunInfo struct {
```

Comment:
there are lots of types here and it's a bit confusing to keep track of all the casting/movement of information which I imagine will make debugging a bit difficult - is there any way we can clean this up? This was mostly inspired by all the different types used in testing/e2e/tests/run.go

Reply:
realistically I'm not sure how exactly this will take place but happy to chat about what my specific pain points are - just a lot of mixes between lists and types that contain lists, similar names, etc etc - I just feel like it can be cleaner/that we can consolidate a lot of the information that is repeatedly moving around

Reply:
it's definitely a lot but using more times makes this simpler in this case. Using separate types to move information up means we can take away information making it easier to develop with.

If you can think of a specific way to improve this I'm happy to implement that in the future but this is the best I have for now.